ops/framework.py

```diff
@@ -1010,6 +1045,70 @@ def __get__(self, parent, parent_type=None):
                 self.__class__.__name__, parent_type.__name__))
 
 
+class ConfigHistoryObserver(Object):
+    """A class used to store config history for current application."""
+    _stored = StoredState()
```

Comment:
Isn't stored state a per-unit construct? Not sure if this is a concern but if an app is scaled up at some point then units would have different history records.

Reply:
Yes and that is right, same think apply for stopped unit.
I think that it's make sense that unit will have different config history, because this will be used mostly when you want to restart some services or close and open port, etc.